hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,799 @@
+            invoice.write({"l10n_mx_edi_usage": data.get("code_usage", "G01")})
+
+        # Confirm invoice
+        invoices.action_post()
+
+        return {
+            "message": "Invoice created",
+            "list_invoices": invoices.read(["name", "date"]),
+        }
+
+    @route(
+        '/register_payment_invoice/<model("account.move"):invoice>',
+        methods=["POST"],
+        type="json",
+        auth="user",
+    )
+    def register_payment(self, invoice=False):
+        """Registers a payment for an invoice.
+
+        This function registers a payment for a specified invoice using the details
+        provided in the JSON request body.
+
+        URL parameter:
+            - invoice (account.move): The invoice model instance.
+
+        JSON request body:
+            - amount (float): The amount to be paid.
+            - journal_id (int): The ID of the payment journal.
+            - payment_method_id (int): The ID of the payment method.
+
+        JSON response:
+            - success (str): A message indicating that the payment has been successfully registered.
+
+        Returns:
+            dict: A dictionary with a success message.
+
+        """
+        data = request.jsonrequest
+        amount = data.get("amount")
+        journal_id = data.get("journal_id")
+        payment_method_id = data.get("payment_method_id")
+
+        register_payment_wizard = request.env["account.payment.register"].with_context(
+            active_model="account.move", active_ids=[invoice.id]
+        )
+
+        wizard = register_payment_wizard.create(
+            {
+                "amount": amount,
+                "journal_id": journal_id,
+                "l10n_mx_edi_payment_method_id": payment_method_id,
+                "communication": "Payment for invoice %s" % invoice.name,
+            }
+        )
+
+        # Register the payment
+        wizard.action_create_payments()
+
+        return {"success": "The payment has been successfully registered."}        
+
+    @route('/get_shipping_info/<model("sale.order"):order>', methods=["GET"], type="json", auth="user")
+    def get_shipping_info(self, order):
+        """Retrieves shipping information for a given sale order.
+
+        This function gathers shipping information related to the specified sale order,
+        including delivery address details and shipping records.
+
+        URL parameter:
+            - order (sale.order): The sale order model instance.
+
+        JSON response:
+            - message (str): A message indicating that the shipping data has been successfully retrieved.
+            - shipping_data (list): A list of dictionaries containing shipping information.
+
+        Returns:
+            dict: A dictionary with a success message and the shipping data.
+        """
+        return order.get_shipping_info()
+
+    @route(
+        "/download_invoice/<int:invoice_id>", methods=["GET"], type="http", auth="user"
```

Comment:
what about to use the model
https://github.com/Vauxoo/addons-vauxoo/blob/15.0/website_sale_brand_catalogs/controllers/portal.py#L10

Reply:
Do I use this model for a new endpoint? @hugho-ad